spec/classes/mysql_backup_mysqldump_spec.rb

```diff
@@ -76,13 +76,13 @@ class { 'mysql::server': }
         let(:params) do
           {
             'file_per_database' => true,
-            'excludedatabases' => ['information_schema']
+            'excludedatabases' => ['information_schema', 'performance_schema']
           }.merge(default_params)
         end
 
         it {
           expect(subject).to contain_file('mysqlbackup.sh').with_content(
-            %r{information_schema},
+            %r{information_schema\\\|performance_schema},
```

Comment:
Aren't that two \ too much?

```
root@webserver01 ~ # mysql -s -r -N -e 'SHOW DATABASES' | grep -v '^\(bastelfreak_blog\\|bastelfreak_past\)$'
bastelfreak_blog
bastelfreak_past
bastelfreak_wp
information_schema
mysql
nextcloud
performance_schema
sys
root@webserver01 ~ # mysql -s -r -N -e 'SHOW DATABASES' | grep -v '^\(bastelfreak_blog\|bastelfreak_past\)$'
bastelfreak_wp
information_schema
mysql
nextcloud
performance_schema
sys
root@webserver01 ~ # mysql -s -r -N -e 'SHOW DATABASES' | grep -v '^\(bastelfreak_blog\\\|bastelfreak_past\)$'
bastelfreak_blog
bastelfreak_wp
information_schema
mysql
nextcloud
performance_schema
sys
root@webserver01 ~ #
```

Or is that just for the escaping due to %r being a regular expression?

Reply:
Yeah, the three escapes are necessesary because we are matching a regular expression with a regular expression.  The `|` and `\` having special meanings on all "layers".